aws/resource_aws_codedeploy_deployment_config.go

```diff
@@ -52,6 +64,72 @@ func resourceAwsCodeDeployDeploymentConfig() *schema.Resource {
 				},
 			},
 
+			"traffic_routing_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								codedeploy.TrafficRoutingTypeAllAtOnce,
+								codedeploy.TrafficRoutingTypeTimeBasedCanary,
+								codedeploy.TrafficRoutingTypeTimeBasedLinear,
+							}, false),
+							Default: codedeploy.TrafficRoutingTypeAllAtOnce,
+						},
+
+						"time_based_canary": {
+							Type:          schema.TypeList,
+							Optional:      true,
+							ForceNew:      true,
+							ConflictsWith: []string{"traffic_routing_config.time_based_linear"},
```

Comment:
For this to work, it needs to reference an element in the list. Its a bummer if this doesn't provide an error currently during resource validation. 😄 

```suggestion
							ConflictsWith: []string{"traffic_routing_config.0.time_based_linear"},
```

Hopefully at some point after Terraform 0.12 ships, we can implement something that allows a relative attribute reference to make these easier to implement. 😖

Reply:
Whoops, forgot to go back in and fix that.